source/SpinalHDL/Libraries/stream.rst

```diff
@@ -489,6 +489,29 @@ or
    val inputStream = Stream(Bits(8 bits))
    val outputStreams = StreamFork(inputStream,portCount=2, synchronous=true)
 
+StreamMux
+^^^^^^^^^^
+A mux implementation for stream, which connect output to input specified by `io.select` signal.
```

Comment:
```suggestion

A mux implementation for stream, which connects output to input specified by `io.select` signal.
```

```suggestion

A mux implementation for ``Stream``. It takes a ``select: UInt`` signal and an ``inputStreams: Vec(Stream)``, and returns a ``Stream`` which is connected to the input specified by ``select``.
```

Reply:
I think the existing style of documentation is more about to show the code by example but not an API Reference.
So the description of those arguments is really not that necessary to show the usage, especially on their types.

On other side, if there is something special need to be mentioned, then it is necessary to explain the example with signal in it.
There are examples explain that begin with "In above example, ....".

There should be a common policy there.

Reply:
Yes, just fixing the typo might be enough (first suggestion). What I didn't understand after first read was the "connect output to input specified by `io.select` signal.": in the example I don't see a `Component` so I cannot find an `io.select`. This I why I suggested the version which is more "arguments to returned value".

> There should be a common policy there.

True

Reply:
> 

emmm, the io.select is in the StreamMux's definition.
So whether use a io signals or signals in example to explain is a problem.